From 85a80d935ebeb5372f9ac4fe497198e7c130b826 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@marvell.com>
Date: Tue, 30 Jul 2019 14:54:14 -0700
Subject: [PATCH 0423/1921] mmc: cavium: do not drop bus lock in tuning

An earlier patch dropped/reacquired bus lock around
hs400 tuning calls. This is unnecessary and deadlock
prone, and was a relic of a hunt for an earlier issue,
since resolved.

With this change, all 3 mmc-slots can be concurrently
active even as some undergo hs200-retuning, without
deadlock.

Change-Id: I42089ea265cf61d088debc3dade2efbffb7f9c46
Signed-off-by: Peter Swain <pswain@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/14925
Reviewed-by: Peter Swain <pswain@cavium.com>
Reviewed-by: Sujeet Kumar Baranwal <Sujeet.Baranwal@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/26918
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/mmc/host/cavium.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index 8701db4ea8f3..9ea58cd5ca41 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -1171,14 +1171,11 @@ static int cvm_mmc_data_tuning(struct mmc_host *mmc, u32 *statp, u32 opcode)
 	struct mmc_card *card = mmc->card;
 
 	if (!(slot->cached_switch & MIO_EMM_SWITCH_HS400_TIMING)) {
-		struct cvm_mmc_host *host = slot->host;
 		int edetail = -EINVAL;
 		int core_opinion;
 
-		host->release_bus(host);
 		core_opinion =
 			mmc_send_tuning(mmc, opcode, &edetail);
-		host->acquire_bus(host);
 
 		/* only accept mmc/core opinion  when it's happy */
 		if (!core_opinion)
-- 
2.31.1

